lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp

```diff
@@ -579,6 +613,7 @@ DWARFASTParserClang::ParseTypeModifier(const SymbolContext &sc,
   LanguageType cu_language = SymbolFileDWARF::GetLanguage(*die.GetCU());
   Type::ResolveState resolve_state = Type::ResolveState::Unresolved;
   Type::EncodingDataType encoding_data_type = Type::eEncodingIsUID;
+  TypePayloadClang payload(GetOwningClangModule(die));
```

Comment:
Why is overwriting the payload for ptrauth fine here? Could it interfere with the modules logic?

In the downstream version of this we leave the payload as is, but instead add the ptrauth qualifiers to the `clang_type` here directly. Did you encounter issues with that approach?

Reply:
I've checked direct usages of the `m_payload` field in `lldb_private::Type`, usages of `SetPayload` and `GetPayload` and also just grepped anything payload-related. As described in comment near `class TypePayloadClang` definition, the payload contains info about owning module ID and the IsCompleteObjCClass flag. The latter does not interfere with overriding payload for ptrauth types since it's only used for ObjC-specific types (while ptrauth types are not). Regarding modules functionality - it's only used for typedefs, which are also not ptrauth types. So, we have several different contexts where payload can be used, and they all are different in terms of type kinds they apply to.

Regarding adding ptrauth qualifiers here directly: in the initial PR https://github.com/apple/llvm-project/pull/8239/, a `FIXME` was left since fully resolving the type here may affect performance. All other types are resolved lazily, and doing the same for ptrauth types makes things consistent. Just like for typedefs, we need to store additional info for ptrauth types to be able to resolve the type on demand later.